Create new Venv for Python-based Project

sharon-wang opened the pull request:
<!-- Thank you for submitting a Pull Request. Please:
* Read our Pull Request guidelines:
  https://connect.posit.it/positron-wiki/pull-requests.html
* Ensure that the code is up-to-date with the `main` branch.
-->

### Intent

- Address https://github.com/posit-dev/positron/issues/2838

#### Demo
https://github.com/posit-dev/positron/assets/25834218/692c5c2d-a236-48aa-903f-52cd4d31be95

### Approach

#### python extension
- add `interpreterPath` to the `CreateEnvironmentOptionsInternal` options so we can skip user input via quickpick, as we'll already have the interpreterPath for a global python installation
- modify `VenvCreationProvider.createEnvironment()` to use `interpreterPath` if provided
- modify `validateMetadata` to return a corresponding registered runtimeMetadata based on the `pythonPath` -- this allows us to pass in a runtimeMetadata object which only contains a `pythonPath` and receive an actual filled-out runtimeMetadata object

#### new project service
- add support for creating a new Python environment using Venv
- install `ipykernel` in the created environment
- autostart the interpreter initialized in the created environment

#### project wizard dialog
- leverage the `pythonEnvProviderName` to omit Conda envs from the project config, since we don't currently support Conda (support will be added via #2839)
- only install `ipykernel` for existing interpreters, since new environments will be handled by the new project service

### QA Notes

Suggested steps
1. Open the New Project Wizard
2. Select Python Project or Jupyter Notebook
3. Project name/location should be inconsequential (defaults are fine)
4. Select New Environment and use Venv as the provider
5. Select any version of Python in the interpreter dropdown
6. Create the Project

Expected outputs
- once the new project window is loaded, a `.venv` directory should be created after a few moments, with `ipykernel` installed
- the Venv interpreter (based on the selected version of Python in the project wizard) should be selected as the active interpreter
- a blank Python/Jupyter file should be open
- the Console should be functional and reflect the Venv interpreter

sharon-wang committed:
add `interpreterPath` to create env options in python extension

sharon-wang committed:
add checks for conda when setting env provider

Conda is not yet supported, so don't allow it as a python env provider when creating the new project.

sharon-wang committed:
create a Venv and install `ipykernel` in it

Next: start the runtime in the Venv

sharon-wang committed:
only pre-install `ipykernel` for existing python environments

sharon-wang committed:
start the new env interpreter

jmcphers reviewed (approved):
LGTM modulo a couple of questions.

sharon-wang committed:
organize positronNewProjectService into code regions

sharon-wang committed:
create the new blank files before env tasks

Env tasks can take a while, but creating the blank files is quick. This way, the blank files are interact-able while the env is set up in the background.

sharon-wang committed:
log when we hydrate metadata in `doAutoStartRuntime`

sharon-wang committed:
add user notifications and improve logging in new project service
